service/history/handler/handler.go

```diff
@@ -1601,6 +1601,9 @@ func (h *handlerImpl) GetReplicationMessages(
 
 		size := proto.FromReplicationMessages(tasks).Size()
 		if (responseSize + size) >= maxResponseSize {
+			metricsScope := h.GetMetricsClient().Scope(metrics.ReplicationMessageTooLargeScope, metrics.ShardIDTag(int(shardID)))
```

Comment:
let's reuse the metric scope returned in line 1558

Reply:
changed. I only added the shard tag inside of that if statement.